JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/ProjectBean.java

```diff
@@ -1,211 +1,282 @@
+
+        //-----------------------------------------set information related to project
+        project.setProjectName(projectName);
+        project.setGroup(group);
+        //-----------------------------------------set hash
+        String projectHash = getHash(project.getProjectId());
+        hashes.setHash(projectHash);
+        project.setHash(hashes);
+        hashes.setProject(project);
+        em.persist(project);
+        //em.flush();
+        System.out.println("New hash id is"+hashes.getId());
+        addProject(group,project);
+        System.out.println("The number of projects in this group is  " + group.getProjects().size());
+
+        return project;
+
+    }
+
+    private Hash getHashFromBase(int projectId) throws UnsupportedEncodingException, NoSuchAlgorithmException {
+        String Hash = getHash(projectId);
+        Hash h;
+
+        try {
+            h = (Hash)em.createQuery("SELECT h FROM Hash h WHERE h.hash=:hash")
+                    .setParameter("hash", Hash)
+                    .getSingleResult();
+        } catch (NoResultException noResult) {
+
+            logger.log(Level.WARNING, "NO RESULT IN QUERY", noResult);
+            return null;
         }
 
-//        Group group;
-//
-//
-//        Project project = new Project();
-//        Hash hashes = new Hash();
-//
-//        //get group, corresponding to this id
-//        group = getGroup(groupId);
-//
-//        //-----------------------------------------set information related to project
-//        project.setProjectName(projectName);
-//        project.setGroup(group);
-//
-//        em.getTransaction().begin();
-//        em.persist(project);
-//        em.getTransaction().commit();
-//
-//        project = getProject(projectName, groupId);
-//
-//        //------------------------------------------set information related to hashes
-//        //  String projectHash = getHash(projectName);
-//        hashes.setProject(project);
-//        // hashes.setHash(projectHash);
-//
-//        em.getTransaction().begin();
-//        em.persist(hashes);
-//        em.getTransaction().commit();
-
-
-
-
-        return null;
+        return h;
 
     }
 
-    public void renameProject(String projectName, int groupId) {
+    private void updateHashFromBase(int projectId, Project p) throws UnsupportedEncodingException, NoSuchAlgorithmException {
 
-        Project project;
-        project = getProject(projectName, groupId);
+        String Hash = getHash(projectId);
 
-        project.setProjectName(projectName);
+        Hash h = (Hash)em.createQuery("SELECT h FROM Hash h WHERE h.hash=:hash")
+                .setParameter("hash", Hash)
+                .getSingleResult();
+
+        h.setProject(p);
+        em.persist(h);
 
-        em.getTransaction().begin();
-        em.persist(project);
-        em.getTransaction().commit();
     }
 
-    public void deleteProject(String projectName, int groupId) {
 
+    private Group getGroup(String groupName) {
```

Comment:
I simply didn't want to create an unnesessary instance of GroupBan there, as there will be some interference. 
23 марта 2016 г., в 2:38, alexander1leonov <notifications@github.com> написал(а):

> In JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/ProjectBean.java:
> 
> >  
> > +    private Group getGroup(String groupName) {
> Avoid copypasting code. Reuse it instead.
> This is already present in Group bean.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>